Automattic/jetpack
# RNA Connection component

Partially extracted from #18784 and #18182.
Previous step: #19134

#### Changes proposed in this Pull Request:
- RNA Connection package now fully encapsulates the connection flow.
- Connection Manager (`connection-ui`) uses RNA Connection to showcase the connection flow.

#### Jetpack product discussion
p1HpG7-bEn-p2

#### Does this pull request change what data or activity we track or use?
No.

#### Testing instructions:
_It's easier to use Jetpack Monorepo to test this PR._
_Make sure to run `yarn build` from the repo root after checking out the branch._

1. Disconnect Jetpack, connect Jetpack using In-Place flow. Confirm that you get connected successfully, and disconnect it again.
2. Go to "Tools -> Connection Manager", click the "Connect" button. After a little wait for the API request, line "Site Registered" should appear, and the In-Place iframe should show up.
3. Click "Approve" in the iframe. The connection form should disappear, and the status line should say "Site and User Connected".
4. Go to Jetpack and disconnect.
5. Set the constant `define( 'JETPACK_SHOULD_NOT_USE_CONNECTION_IFRAME', true )` to test the Calypso flow.
6. Go back to "Tools -> Connection Manager" and click the "Connect" button. After a little wait, you should get redirected to Calypso.
7. Complete the connection flow. You will be redirected to Jetpack Dashboard and not the Connection Manager. That's a known issue and will be addressed later.
8. Disconnect Jetpack again, remove the `JETPACK_SHOULD_NOT_USE_CONNECTION_IFRAME` constant.
9. Go to "Tools -> Connection Manager" and click "Connect". Confirm that the In-Place iframe and the line "Site Registered" appear. Then reload the page.
10. After reloading the page, you should see the "Site Registered" line and the "Connect" button. Click the button and finish the connection flow.
